Fraser/1.99.3/fraser.R

```diff
@@ -0,0 +1,158 @@
+  if(dir.exists(input)){
+    sampleID <- list.files(path=input,pattern="*.bam",full.names=FALSE)
+    bamFile <- list.files(path=input,pattern="*.bam",full.names=TRUE)
+    return(data.table(sampleID,bamFile))
+  }  
+  else if(file.exists(input)){
+    sampleID <- basename(file.path(input))
+    bamFile <- file.path(input)
+    return(data.table(sampleID,bamFile))
+  }
+}
+
+
+create_sample_table <- function(sample,ref,paired_end){
+  sampleTable <- rbind(get_st_rows(sample),get_st_rows(ref))
+  sampleTable <- sampleTable[!grepl(".bai",sampleID), ]
+  sampleTable$group <- c(1:nrow(sampleTable))
+  sampleTable$pairedEnd <- TRUE
+
+  message(date(), ": sampleTable")
+  print(sampleTable)
+
+  return(sampleTable)
+}
+
+
+run_fraser <- function(sampleTable){
+  # create FRASER object
+  settings <- FraserDataSet(colData=sampleTable, workingDir="./FRASER_output")
+  message(date(), ": settings...")
+  print(settings)
+
+  register(MulticoreParam(workers=min(8, multicoreWorkers())))
+
+  # count reads
+  fds <- countRNAData(settings)
+
+  fds <- calculatePSIValues(fds)
+
+  #FRASER vignette/gagneurlab settings:
+  #Currently, we suggest keeping only junctions which support the following:
+  #minExpressionInOneSample - 20 - At least one sample has 20 (or more) reads
+  #quantileMinExpression - 10 - The minimum total read count (N) an intron needs to have at the specified quantile across samples to pass the filter. See quantileForFiltering. 
+  #quantile - 0.95 -  Defines at which percentile the quantileMinExpression filter is applied. 
+  #A value of 0.95 means that at least 5% of the samples need to have a total read count N >= quantileMinExpression to pass the filter.
+  #minDeltaPsi - 0.05 -  The minimal variation (in delta psi) required for an intron to pass the filter. gagneurlab suggests 0.05
+
+  fds <- filterExpressionAndVariability(fds, minExpressionInOneSample = 20, quantileMinExpression=10, quantile = 0.95, minDeltaPsi=0.05, filter=FALSE)
+  fds <- saveFraserDataSet(fds, name="fds-filter")
+  plotheat = "plots_fraser.pdf"
+  pdf(plotheat,onefile = TRUE)
+  plotFilterExpression(fds, bins=100)
+
+  message(date(), ": dimension before filtering...")
+  print(dim(assays(fds)$rawCountsJ))
+  print(dim(assays(fds)$rawCountsSS))
+
+  fds <- fds[mcols(fds, type="j")[,"passed"]]
+  message(date(), ": dimension after filtering...")
+  print(dim(assays(fds)$rawCountsJ)) 
+  print(dim(assays(fds)$rawCountsSS))
+
+  #Finds the optimal encoding dimension by injecting artificial splicing outlier ratios while maximizing the precision-recall curve.
+  # Get range for latent space dimension, max tested dimensions = 6
+  mp <- 6 
+  a <- 2
+  b <- min(ncol(fds), nrow(fds)) / mp   # N/mp
+
+  maxSteps <- 12
+  if(mp < 6){
+    maxSteps <- 15
+  }
+
+  Nsteps <- min(maxSteps, b)
+  pars_q <- unique(round(exp(seq(log(a),log(b),length.out = Nsteps))))
+
+  message(date(), ": getEncDimRange", pars_q)
+
+  psiType = "jaccard"
+  impl = "PCA"
```

Comment:
It is preferable to set variables that won't change like these as a constant in the top of your script or pass them as an argument to your script